components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -17,8 +17,10 @@ namespace ntp_background_images {
 ViewCounterModel::ViewCounterModel(PrefService* prefs) : prefs_(prefs) {
   CHECK(prefs);
 
+  // When browser is restarted or component is updated, we reset to "initial"
+  // count.
   count_to_branded_wallpaper_ =
```

Comment:
I think we have to reset this value from `ViewCounterModel::Reset()`.
It's called whenever component is updated.